### ISSUE TITLE
Problems with extends:

### ISSUE THREAD

**Issue by iandol** [htdebeer/pandocomatic#33]
This may be a bit difficult to track down and I currently have very little free time to try to debug my way through this. My pandocomatic YAML is here:

https://github.com/iandol/dotfiles/blob/master/pandoc/pandocomatic.yaml

What seems to be happening is that if I try to use a parent template, in this case [`latex-with-refs`](https://github.com/iandol/dotfiles/blob/master/pandoc/pandocomatic.yaml#L17), then actually all the settings of the templates that extend it are also included? So I get this debug output from pandocomatic:

Brain-AI.md:
```
pandocomatic:
  use-template:
    - latex-with-refs
```

Debug output:
```
👉  pandocomatic -b Brain-AI.md
pandoc	--from=markdown \
	--to=latex \
	--standalone \
	--filter=/usr/local/bin/pandoc-citeproc \
	--filter=/Users/ian/.pandoc/filters/removeHR \
	--filter=/Users/ian/.pandoc/filters/authorToName \
	--bibliography=/Users/ian/.pandoc/Core.bib \
	--csl=/Users/ian/.pandoc/csl/apa-ian.csl \
	--template=/Users/ian/.pandoc/templates/custom-memoir.latex \
	--dpi=300 \
	--table-of-contents \
	--base-header-level=1 \
	--latex-engine=xelatex \
	--toc-depth=4 \
	--output=./Brain-AI.pdf
```

Even though the output is supposed to be `latex` I get an attempt to build a PDF ([probably from this template](https://github.com/iandol/dotfiles/blob/master/pandoc/pandocomatic.yaml#L61)), and the `pandoc:template` is not the template of `latex-with-refs` but of [`memoir-with-refs`](https://github.com/iandol/dotfiles/blob/master/pandoc/pandocomatic.yaml) which is the last template to inherit from `latex-with-refs`. I can't see the metadata in the debug output, but I assume it is merged as well.

Sorry not to have a simpler test case example...

**Comment by htdebeer** [comment]
Don't worry. I'll look into this issue tomorrow.

**Comment by htdebeer** [comment]
Found the issue: I forgot to deep copy template Hashes. Fixed in version 0.1.4.11.

**Comment by iandol** [comment]
I can confirm this is fixed and this has really made my [pandocomatic.yaml](https://github.com/iandol/dotfiles/blob/master/pandoc/pandocomatic.yaml) so much easier to maintain, brilliant!!!
